helper-cli/src/main/kotlin/commands/dev/RewriteTestAssetsCommand.kt

```diff
@@ -0,0 +1,207 @@
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.EvaluatorRun
+import org.ossreviewtoolkit.model.FileFormat
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScannerRun
+import org.ossreviewtoolkit.utils.common.expandTilde
+
+internal class RewriteTestAssetsCommand : CliktCommand(
+    help = "Searches all test assets directories in the given ORT sources directory for recognized serialized files " +
+            "and tries to de-serialize and serialize the file. The command can be used to update the test assets " +
+            "after making changes to the corresponding model classes or serializer configuration, e.g. after " +
+            "annotating a property to not be serialized if empty."
+) {
+    private val ortSourcesDir by option(
+        "--ort-sources-dir", "-i",
+        help = "The directory containing the source code of ORT."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = false, canBeDir = true, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val candidateFiles = findCandidateFiles(ortSourcesDir)
+        println("Found ${candidateFiles.size} candidate file(s).\n")
+
+        val candidateFilesByType = patchAll(candidateFiles)
+
+        candidateFilesByType.forEach { (type, files) ->
+            files.forEach { file ->
+                println("[$type] ${file.absolutePath}")
+            }
+        }
+
+        println("\nSummary\n")
+
+        candidateFilesByType.forEach { (type, files) ->
+            println("$type: ${files.size}")
+        }
+    }
+}
+
+// Fake values to replace the replace patterns with in order to make de-serialization work. The fake values must be
+// unique and no fake value must be a substring of another.
+private val REPLACE_PATTERN_REPLACEMENTS = listOf(
+    "<REPLACE_JAVA>" to "33445501",
+    "<REPLACE_OS>" to "33445502",
+    "\"<REPLACE_PROCESSORS>\"" to "33445503",
+    "\"<REPLACE_MAX_MEMORY>\"" to "33445504",
+    "<REPLACE_DEFINITION_FILE_PATH>" to "33445505",
+    "<REPLACE_ABSOLUTE_DEFINITION_FILE_PATH>" to "33445506",
+    "<REPLACE_URL>" to "http://replace/url/non/processed",
+    "<REPLACE_REVISION>" to "33445507",
+    "<REPLACE_PATH>" to "replace/path",
+    "<REPLACE_URL_PROCESSED>" to "http://replace/url/processed",
+)
+
+private val TARGET_CLASSES = setOf(
+    AnalyzerResult::class,
+    AnalyzerRun::class,
+    EvaluatorRun::class,
+    OrtResult::class,
+    ScannerRun::class,
+    ScanResult::class,
+    ProjectAnalyzerResult::class
+)
+
+// Paths to nodes in the tree of JsonNodes, whose subtree shall not be changed. Explicitly ignoring these subtrees is
+// needed in order to avoid bringing in default values property values which are not present in the original file.
```

Comment:
> what I believe you mean is the *machine-specific values* of the config and environment classes

Yet more detailed, if machine-specific values would be used, the replacing back of dummy values to placeholders would not work, right?

Reply:
> Yet more detailed, if machine-specific values would be used, the replacing back of dummy values to placeholders would not work, right?

I don't understand this question. Can you elaborate a bit?

Reply:
My thinking was that you first replace e.g.

    `"\"<REPLACE_MAX_MEMORY>\""` with `"33445504"`

and later again replace

    `"33445504"` with `"\"<REPLACE_MAX_MEMORY>\""`

and in between those two replacements `"33445504"` would get replaced with the real amount of memory of the machine running the command, the back-replacing would not work anymore. But giving it a second thought, I believe this cannot happen, exactly because these values are being patched, i.e. they are present in the file, and thus no default value is used for them.

Reply:
> I'm not sure what you exactly refer to with "machine-specific"

I'm referring to default values, esp. in `Environment`, that vary from machine to machine, like the `maxMemory` property. I had a feeling that the real reason for the code is that you only want to not take those properties. But if you really also do not want to take properties like `enableRepositoryPackageConfigurations`, then the term "machine-specific" obviously is not applicable.